arbor/backends/gpu/stack.hpp

```diff
@@ -110,29 +150,32 @@ class stack {
 
     value_type& operator[](unsigned i) {
```

Comment:
Maybe we can deprecate all of these `operator[]` and `begin, end` if we are now providing a function that returns a `const std::vector<T>&` with the values.

Reply:
Sure. But it doesn't seem to be a maintenance overhead as that bit is unlikely to change.